### PR TITLE
distutils_install_for testing install proper egg-info, dev-python/cryptography fix test on first install

### PR DESCRIPTION
**Pull request by MathyV**
Currently distutils_install_for_testing does not install the complete
egg-info into ${TEST_DIR}. This was first noticed by W. Trevor King and
reported in bug #524322. Based on info found in the related upstream
setuptools bug I added the necessary call to setuptools to create the
complete egg_info.

Without this certain packages (like cryptography) fail during testing
because they use introspection but the metadata isn't available.

Gentoo-Bug: https://bugs.gentoo.org/show_bug.cgi?id=595272
Gentoo-Bug: https://bugs.gentoo.org/show_bug.cgi?id=524322

### CODE REVIEW

File: eclass/distutils-r1.eclass [hunk 1]
```diff
@@ -287,15 +287,21 @@ distutils_install_for_testing() {
...
 
+	local add_args_egg=(
+		egg_info
+			--egg-base="${libdir}"
+	)
+
 	mkdir -p "${libdir}" || die
-	esetup.py "${add_args[@]}" "${@}"
+	esetup.py "${add_args_egg[@]}"
+	esetup.py "${add_args_install[@]}" "${@}"
```
Comment (mgorny) [inline eclass/distutils-r1.eclass]: Any reason not to pass both in a single call?

**Comment by mgorny** [comment]
CC @gentoo/python.

**Comment by floppym** [comment]
I'm a bit confused as to why `esetup.py install` does not copy/create the egg-info. But if this fixes the problem, I suppose that's good enough for me.

File: eclass/distutils-r1.eclass (hunk 1 above)
Comment (MathyV) [inline eclass/distutils-r1.eclass]: Other than for the sake of clarity... no I guess?

**Comment by MathyV** [comment]
@floppym the install puts in some egg-info but not everything and for some reason this seems to be desired behaviour...

**Pull request closed without merging**